### ISSUE TITLE
Global Middleware Fails

### ISSUE THREAD

**Issue by ssljivic** [pleerock/routing-controllers#91]
When MiddlewareGlobalBefore is used, the following error is thrown:

> 
> TypeError: Cannot read property 'headers' of undefined
>     at ExpressDriver.handleError (C:\\app\node_modules\routing-controlrs\driver\ExpressDriver.js:273:31)
>     at C:\\app\node_modules\routing-controllers\driver\ExpressDriver.js:75:23
>     at Layer.handle [as handle_request] (C:\\app\node_modules\express\lib\router\layer.js:95:5)
>     at trim_prefix (C:\\app\node_modules\express\lib\router\index.js:317:13)
>     at C:\\app\node_modules\express\lib\router\index.js:284:7
>     at Function.process_params (C:\\app\node_modules\express\lib\router\index.js:335:12)
>     at next (C:\\app\node_modules\express\lib\router\index.js:275:10)
>     at expressInit (C:\\app\node_modules\express\lib\middleware\init.js:40:5)
>     at Layer.handle [as handle_request] (C:\\app\node_modules\express\lib\router\layer.js:95:5)
>     at trim_prefix (C:\\app\node_modules\express\lib\router\index.js:317:13)

Package versions:

- routing-controllers: 0.6.10
- express: 4.15.0

**Comment by ssljivic** [comment]
The problem is that in ExpressDriver.ts:81 you have:

```
this.handleError(error, undefined, {
                    request: request,
                    response: response,
                    next: next,
                });
```

And in function `handleError(error: any, action: ActionMetadata, options: ActionCallbackOptions): void {` on line 292, we have:

```
Object.keys(action.headers).forEach(name => {
                response.header(name, action.headers[name]);
            });
```

And since action is explicitly passed as undefined, we have an error.

**Comment by pleerock** [comment]
fixed in `0.6.12`
